security/oc_pki.c

```diff
@@ -362,6 +365,58 @@ oc_pki_add_trust_anchor(size_t device, const unsigned char *cert,
   return pki_add_trust_anchor(device, cert, cert_size, OC_CREDUSAGE_TRUSTCA);
 }
 
+void
+oc_pki_set_verify_certificate_cb(oc_pki_verify_certificate_cb_t cb)
+{
+  g_verify_certificate_cb = cb;
+}
+
+static int
+default_verify_certificate_cb(struct oc_tls_peer_t *peer, mbedtls_x509_crt *crt,
+                              int depth, uint32_t *flags)
+{
+  (void)peer;
+  (void)crt;
+  (void)depth;
+  (void)flags;
+  if (flags != NULL && (*flags & MBEDTLS_X509_BADCERT_EXPIRED) != 0) {
+    char buf[256];
+    int ret = mbedtls_x509_dn_gets(buf, sizeof(buf) - 1, &crt->subject);
+    if (ret >= 0) {
+      buf[ret] = 0;
+    } else {
+      ret = snprintf(buf, sizeof(buf) - 1, "unknown");
+      buf[ret] = 0;
+    }
+    OC_ERR("certificate %s is expired on %d-%02d-%02d %02d:%02d:%02d", buf,
```

Comment:
You could put lines 382-407 under

```C
#ifdef OC_DEBUG
if (flags != NULL && (*flags & MBEDTLS_X509_BADCERT_EXPIRED) != 0) {
...
}
#endif /* OC_DEBUG */
```

to save some bytes, because `OC_DBG`, `OC_WRN` and `OC_ERR` are defined as nothing when `OC_DEBUG` is not defined (see oc_log.h). In fact, I'm surprised that the compiler didn't complain about unused variables and fail.

Reply:
fixed